### PR TITLE
Add Android Version in app list

### PR DESCRIPTION
**Pull request by HO-COOH**
I am a new Android developer just getting my feet wet. I made this PR to show Android version alongside with API version in both the app list and app details, because api levels are kind of messy to remember. (I am aware there is Android versions in the chart view, but show it in the app list is handy to me)
Screenshot_20220702-164919_LibChecker

The text is getting a bit long tho. If you are not satisfied, maybe we can work together to add a switch like only showing api levels/only showing Android versions/showing both.

### CODE REVIEW

**Comment by zhaobozhen** [comment]
Thanks for submitting the PR, but I'd like to keep the front page of the app (i.e. the first page the user opens the app to display) relatively clean. I would like to display more information on the details page of the app, so I will consider how to design the API version and Android version against a realistic strategy in a subsequent release.

**Comment by qhy040404** [comment]
How about adding this preference to the settings?

**Comment by qhy040404** [comment]
EXAMPLE:
GlobalValues.kt
`var targetMode: String by SPDelegates(Constants.PREF_TARGET_DISPLAY_MODE, ...)`

The rest is in the review.

File: app/src/main/kotlin/com/absinthe/libchecker/utils/PackageUtils.kt [hunk 1]
```diff
@@ -152,6 +153,11 @@ object PackageUtils {
     return "$versionName ($versionCode)"
   }
 
+  private fun apiLevelToAndroidVer(apiVersion: Int) =
+    if (apiVersion == 0 || apiVersion >= apiToAndroidVerMap.size) ""
+    else "Android${apiToAndroidVerMap[apiVersion]}"
```
Comment (qhy040404) [inline app/src/main/kotlin/com/absinthe/libchecker/utils/PackageUtils.kt:158]:
```
else when (GlobalValues.targetMode) {
  "ver" -> "Android ${apiToAndroidVerMap[apiVersion]}"
  else -> "API $targetSdkVersion" // include "api"
}

**Review by qhy040404 — COMMENTED** [review]
How about adding this preference to the settings?

EXAMPLE:
GlobalValues.kt
`var targetMode: String by SPDelegates(Constants.PREF_TARGET_DISPLAY_MODE, ...)`

Just a suggestion.

File: app/src/main/kotlin/com/absinthe/libchecker/utils/PackageUtils.kt (hunk 1 above)
Comment (qhy040404) [reply app/src/main/kotlin/com/absinthe/libchecker/utils/PackageUtils.kt:158]: and the function name `apiLevelToAndroidVer` -> `getTargetString`

**Comment by zhaobozhen** [comment]
> How about adding this preference to the settings?
> 
> EXAMPLE:
> GlobalValues.kt
> `var targetMode: String by SPDelegates(Constants.PREF_TARGET_DISPLAY_MODE, ...)`
> 
> Just a suggestion.

good idea.  Maybe I can implement an advanced menu page with more configurable items.  I already have some ideas for this, maybe it will be implemented in the next few releases.

**Comment by zhaobozhen** [comment]
> How about adding this preference to the settings?
> 
> EXAMPLE:
> GlobalValues.kt
> `var targetMode: String by SPDelegates(Constants.PREF_TARGET_DISPLAY_MODE, ...)`
> 
> Just a suggestion.

DLUT 的吗，我们不会是校友吧🌚

**Comment by qhy040404** [comment]
> > How about adding this preference to the settings?
> > EXAMPLE:
> > GlobalValues.kt
> > `var targetMode: String by SPDelegates(Constants.PREF_TARGET_DISPLAY_MODE, ...)`
> > Just a suggestion.
> 
> DLUT 的吗，我们不会是校友吧🌚

卧槽这都能碰上，但是我不是软工的 :joy:

**Review by DeadSamurai8 — APPROVED** [review]

File: app/src/main/kotlin/com/absinthe/libchecker/utils/PackageUtils.kt [hunk 2]
```diff
@@ -27,6 +27,7 @@ import com.absinthe.libchecker.bean.KotlinToolingMetadata
 import com.absinthe.libchecker.bean.LibStringItem
 import com.absinthe.libchecker.bean.StatefulComponent
 import com.absinthe.libchecker.compat.PackageManagerCompat
+import com.absinthe.libchecker.constant.AndroidVersions.apiToAndroidVerMap
```
Comment (DeadSamurai8) [inline app/src/main/kotlin/com/absinthe/libchecker/utils/PackageUtils.kt:30]: []()

**Comment by DeadSamurai8** [comment]
> I am a new Android developer just getting my feet wet. I made this PR to show Android version alongside with API version in both the app list and app details, because api levels are kind of messy to remember. (I am aware there is Android versions in the chart view, but show it in the app list is handy to me)
> Screenshot_20220702-164919_LibChecker
> 
> The text is getting a bit long tho. If you are not satisfied, maybe we can work together to add a switch like only showing api levels/only showing Android versions/showing both.
>

**Comment by qhy040404** [comment]
@zhaobozhen This PR should be closed as the feature has already been merged into `master` when 2.3.0. It's duplicate here.

**Pull request closed without merging**